### DISCUSSION TITLE
WNS: Blocked channel URL

### DISCUSSION THREAD

**Discussion by cschwarz** [microsoft/WindowsAppSDK#1292] [Q&A] [answered]
Hi,

we are using WNS for Windows push notifications, but in a certain network environment and only on a few PCs we get back "Blocked channel URL" if we send a push notification. 

We whitelisted the IP ranges according to https://docs.microsoft.com/en-us/windows/apps/design/shell/tiles-and-notifications/firewall-allowlist-config.

I haven't found much infos beside: 

https://stackoverflow.com/questions/32207565/push-message-for-windows-phone-wns-blocked-channel-url
https://social.msdn.microsoft.com/Forums/azure/en-US/8338a90d-93a2-4627-9629-d395b3e7ad99/uwp-c-windows-when-on-battery-saver-drops-push-notifications?forum=wpdevelop

I tried enabling battery saving mode, but I still receive push notifications on my machine.

Are there any more details available to this specific error?

**Comment by btueffers** [comment]
@nicuparente do you have any advice for this situation?

**Comment by nicuparente** [comment, marked as answer]
Hi @cschwarz - can you clarify if you are trying to block the notifications or allow it? "but I still receive push notifications on my machine".

To help with troubleshooting can you provide a MS-CV of the most recent from the WNS HTTP Response Header?  
The "Blocked channel URL" can also be that there is not RAW notification handler for the notification. What is the notification type you are sending? raw, toast, badge or tiles?

**Reply by cschwarz** [reply]
Hi @nicuparente ,

we are trying to allow it (toast and raw notifications), the test was to see if the battery saving mode could cause an issue, as described in the second link.

We have enabled logging of the response headers, but couldn't reproduce the "Blocked channel URL", which only occurred recently (while the push never worked on these machine).

We found the following logs in the event viewer under PushNotifications-Platform -> Operational on that machine as soon as we start our app:

"WNP Transport Layer for data link received asynchronous connection error 10060."

**Reply by nicuparente** [reply]
@cschwarz - I appreciate the patience. This slipped through the cracks.

"WNP Transport Layer for data link received asynchronous connection error 10060."
This issue occurs when we receive a timeout for our connections from our client but this shouldn't cause the channel to get blocked. 

To troubleshoot this further can you provide the most recent MS-CV of a block channel request?
